collated/main/A0139552B.md

```diff
@@ -0,0 +1,254 @@
+        if (taskNameString.isPresent() && !taskNameString.get().toString().trim().equals("")) {
+    		taskName = new Name(taskNameString.get());
+        }
+    }
+
+    private void assignStartDateIfPresent(Optional<String> startDateString) {
+        if (startDateString.isPresent()) {
+            startDate = DateTime.convertStringToDate(startDateString.get());
+            if (!DateTime.hasTimeValue(startDateString.get())) {
+                startDate = DateTime.setTimeToStartOfDay(startDate);
+            }
+        }
+    }
+
+    private void assignEndDateIfPresent(Optional<String> endDateString) {
+        if (endDateString.isPresent()) {
+            endDate = DateTime.convertStringToDate(endDateString.get());
+            if (startDate != null && !DateTime.hasDateValue(endDateString.get())) {
+                endDate = DateTime.setEndDateToStartDate(startDate, endDate);
+            }
+            if (!DateTime.hasTimeValue(endDateString.get())) {
+                endDate = DateTime.setTimeToEndOfDay(endDate);
+            }
+        }
+    }
+
+    private void assignRecurrenceRateIfPresent(Optional<String> rateString, Optional<String> timePeriodString)
+            throws IllegalValueException {
+        if (rateString.isPresent() && timePeriodString.isPresent()) {
+            recurrenceRate = new RecurrenceRate(rateString.get(), timePeriodString.get());
+        } else if (!rateString.isPresent() && timePeriodString.isPresent()) {
+            recurrenceRate = new RecurrenceRate(STRING_CONSTANT_ONE, timePeriodString.get());
+        } else if (rateString.isPresent() && !timePeriodString.isPresent()) {
+            throw new IllegalValueException(RecurrenceRate.MESSAGE_VALUE_CONSTRAINTS);
+        }
+    }
+
+    /*
+     * Assign priority depending on the level stated
+     * Otherwise leave it as null
+     */
+    private void assignPriority(Optional<String> priorityString) {
+        if (priorityString.isPresent()) {
+            switch (priorityString.get()) {
+                case ("low"): case ("l"): priority = Priority.LOW; break; 
+                case ("high"): case ("h"): priority = Priority.HIGH; break;
+                case ("medium"): case ("m"): case ("med"): priority = Priority.MEDIUM; break;
+            }
+        }
+    }
+	
+    /*
+     * Check which field is to be reset
+     */
+    private void fieldsToReset(String resetFieldString) {
+        if(resetFieldString != null){
+        	String[] resetField = resetFieldString.trim().split(" ");
+        	for(int i = 0; i < resetField.length; i++){
+        		switch (resetField[i].trim()) {
+            		case ("repeat"):  removeReccurence = true; break; 
+            		case ("start"): removeStartDate = true; break;
+            		case ("end"): removeEndDate = true; break;
+        		}
+        	}
+        }
+    }
+
+	@Override
+	public CommandResult execute() {	    
+	    assert model != null;
+		
+	    if (model.isCurrentListDoneList()) {
```

Comment:
indentation off in this method in general. please fix it; code formatting is a matter of one key stroke combinations! (which?)